### ISSUE TITLE
Question regarding permutation feature importance metrics implementation

### ISSUE THREAD

**Issue by balazsgonczy** [dreamquark-ai/tabnet#383]
Dear Authors,

I would like to do a simple- and a permutation based feature importance metrics comparison between multiple decision tree algorithms and TABNET. How can I make my code work for TABNET?

Code:

``
def permuted_feature_ranking(classifier, X_train, y_train, model_name="model"):
  import eli5
  from eli5.sklearn import PermutationImportance
  permutation_importance = PermutationImportance(classifier, scoring='neg_mean_squared_error', random_state=0)
  permutation_importance.fit(X_train, y_train)
  df = eli5.explain_weights_df(permutation_importance, feature_names = list(X_train.columns))
  plt.figure(figsize=(20, 10))
  sns.barplot(x="weight", y="feature", data=df.sort_values(by="weight", ascending=False))
  plt.title(f'{model_name} Permuted features importance ranking')
  plt.tight_layout()
  plt.show()
``

The fit part is where it crashes with the following error message: 
"
TypeError: If no scoring is specified, the estimator passed should have a 'score' method. The estimator TabNetClassifier(n_d=64, n_a=64, n_steps=7, gamma=1.8, cat_idxs=[], cat_dims=[], cat_emb_dim=1, n_independent=4, n_shared=4, epsilon=1e-15, momentum=0.05, lambda_sparse=1.9949459631534334e-06, seed=0, clip_value=1, verbose=1, optimizer_fn=<class 'torch.optim.adam.Adam'>, optimizer_params={'lr': 0.02, 'weight_decay': 1e-05}, scheduler_fn=<class 'torch.optim.lr_scheduler.ReduceLROnPlateau'>, scheduler_params={'mode': 'min', 'patience': 21, 'min_lr': 1e-05, 'factor': 0.5}, mask_type='entmax', input_dim=15, output_dim=2, device_name='auto') does not.
"

Best,

Balázs

**Comment by Optimox** [comment]
You'll need to create a wrapping class that expose a `score` method, I don't know what this score corresponds to but the wrapper should be fairly simple and resemble something like the wrapper here : https://github.com/dreamquark-ai/tabnet/issues/146#issuecomment-643825117

**Comment by balazsgonczy** [comment]
I think I have found the solution:

![image](https://user-images.githubusercontent.com/53116701/163436806-9a079bec-6cef-43d0-9fa6-a009141023dc.png)

**Comment by balazsgonczy** [comment]
Can you please have a look at this? Are the two highlighted lines return the same values? Because if yes the I have created a workaround workflow.

![image](https://user-images.githubusercontent.com/53116701/163451762-97023601-25ff-4e12-b435-9794659175f9.png)

**Comment by Optimox** [comment]
yes, permuation importance is based on decrease of scores when doing random permutations on columns.

So if you follow the non sklearn way you'll get what you want.
